### PR TITLE
Add rate rule description

### PR DESCRIPTION
**Pull request by ltachet**
Update the documentation for the `api/2.0/taxes` endpoint.
The rates are now containing rules (e.g.: tax with threshold).

### CODE REVIEW

File: api.json [hunk 1]
```diff
@@ -6939,6 +6939,13 @@
         "display_name": {
           "description": "The name of the tax rate used for display.",
           "type": "string"
+        },
+        "rules": {
+          "type": "array",
+          "description": "Rules that apply to the rate for tax explusive outlets.",
```
Comment (edsrzf) [inline api.json]: "exclusive". 😁  Or "explosive"? 🤔

**Review by edsrzf — APPROVED** [review]
One typo, otherwise LGTM 👍

**Review by pzurek — APPROVED** [review]
👌

**Review by laher — APPROVED** [review]
Looks good to me. Are either of these required fields, which would break existing clients?

Let's do some work on api-testing to verify the behaviour... maybe next week with @Kittyness

**Comment by edsrzf** [comment]
These are new fields that can be returned by the API rather than ones taken as input. Nothing should break existing clients.

**Pull request merged**